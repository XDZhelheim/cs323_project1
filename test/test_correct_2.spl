int test_correct_2()
{
  int a = 100, b = 40;
  while (a != 0) {
    a = a - a / b * b; 
  }
  return b;
}