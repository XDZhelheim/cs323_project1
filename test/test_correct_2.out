Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: test_correct_2
        LP
        RP
      CompSt (2)
        LC
        DefList (3)
          Def (3)
            Specifier (3)
              TYPE: int
            DecList (3)
              Dec (3)
                VarDec (3)
                  ID: a
                ASSIGN
                Exp (3)
                  INT: 100
              COMMA
              DecList (3)
                Dec (3)
                  VarDec (3)
                    ID: b
                  ASSIGN
                  Exp (3)
                    INT: 40
            SEMI
        StmtList (4)
          Stmt (4)
            WHILE
            LP
            Exp (4)
              Exp (4)
                ID: a
              NE
              Exp (4)
                INT: 0
            RP
            Stmt (4)
              CompSt (4)
                LC
                StmtList (5)
                  Stmt (5)
                    Exp (5)
                      Exp (5)
                        ID: a
                      ASSIGN
                      Exp (5)
                        Exp (5)
                          ID: a
                        MINUS
                        Exp (5)
                          Exp (5)
                            Exp (5)
                              ID: a
                            DIV
                            Exp (5)
                              ID: b
                          MUL
                          Exp (5)
                            ID: b
                    SEMI
                RC
          StmtList (7)
            Stmt (7)
              RETURN
              Exp (7)
                ID: b
              SEMI
        RC
